qutebrowser/browser/shared.py

```diff
@@ -339,3 +343,37 @@ def netrc_authentication(url, authenticator):
     authenticator.setPassword(password)
 
     return True
+
+
+def choose_file(multiple: bool) -> List[str]:
+    """Select file(s) for uploading, using external command defined in config.
+
+    Args:
+        multiple: Should selecting multiple files be allowed.
+
+    Return:
+        A list of selected file paths, or empty list if no file is selected.
+        If multiple is False, the return value will have at most 1 item.
+    """
+    encoding = sys.getfilesystemencoding()
+    with tempfile.NamedTemporaryFile(mode='w+', encoding=encoding) as tmpfile:
+        if multiple:
+            command = config.val.fileselect.multiple_files.command
+        else:
+            command = config.val.fileselect.single_file.command
+
+        proc = guiprocess.GUIProcess(what='choose-file')
+        proc.start(command[0],
+                   [arg.replace('{}', tmpfile.name) for arg in command[1:]])
+
+        loop = qtutils.EventLoop()
+        proc.finished.connect(lambda _code, _status: loop.exit())
+        loop.exec_()
+
+        selected_files = tmpfile.read().splitlines()
+
+    if not multiple:
+        if len(selected_files) > 1:
+            log.misc.exception("More than one files choosen, using only the first")
```

Comment:
```suggestion
            message.warning("More than one files choosen, using only the first")
```

So that the warning actually gets displayed in the user interface.

Reply:
Ah nice!